File: 2022/day-01/main.py
from itertools import groupby
import sys

with open(sys.argv[1], 'r') as f:
    lines = [line.strip() for line in f.readlines()]
    elves = [sum(map(int, elf))
             for value, elf in groupby(lines, key=bool) if value]

    answer = max(elves)
    print(f"part 1: {answer}")

    elves.sort(reverse=True)
    answer = sum(elves[0:3])
    print(f"part 2: {answer}")

File: 2022/day-01/src/main.rs
const INPUT: &'static str = include_str!("../input.txt");

fn main() {
    let mut elves = INPUT
        .lines()
        .collect::<Vec<&str>>()
        .split(|s| s.is_empty())
        .map(|s| s.iter().map(|v| v.parse::<usize>().unwrap()).sum())
        .collect::<Vec<usize>>();

    elves.sort();
    elves.reverse();

    println!("part 1: {}", elves[0]);
    println!("part 2: {}", elves.iter().take(3).sum::<usize>());
}
